cmd/nodepool/azure/create.go

```diff
@@ -54,6 +64,21 @@ func (o *AzurePlatformCreateOptions) UpdateNodePool(_ context.Context, nodePool
 		return fmt.Errorf("resource-group-name is required when using disk-encryption-set-id")
 	}
 
+	// Validate publisher, offer, sku, and version flags are provided if one of them is not empty
+	marketplaceImageInfo := map[string]string{
+		"marketplace-publisher": o.MarketplacePublisher,
+		"marketplace-offer":     o.MarketplaceOffer,
+		"marketplace-sku":       o.MarketplaceSKU,
+		"marketplace-version":   o.MarketplaceVersion,
+	}
+	if marketplaceImageInfo["marketplace-publisher"] != "" || marketplaceImageInfo["marketplace-offer"] != "" || marketplaceImageInfo["marketplace-sku"] != "" || marketplaceImageInfo["marketplace-version"] != "" {
+		for flag, value := range marketplaceImageInfo {
+			if value == "" {
+				return fmt.Errorf("all marketplace flags (i.e. marketplace-publisher, marketplace-offer, marketplace-sku, marketplace-version) are required when using a marketplace image; the following flag was empty: %s", marketplaceImageInfo[flag])
+			}
+		}
+	}
```

Comment:
Could we move this to a separate function validateAzureOptions like we have for aws ? I feel like with the development for Azure we will be wanting more and more validation so makes sense to create a new func for it

Reply:
Makes sense. Will update that.